editor/js/Loader.js

```diff
@@ -697,35 +697,37 @@ function Loader( editor ) {
 
 	async function handleZIP( contents ) {
 
-		var zip = new JSZip( contents );
+		var zip = unzipSync( new Uint8Array( contents ) );
 
 		// Poly
 
-		if ( zip.files[ 'model.obj' ] && zip.files[ 'materials.mtl' ] ) {
+		if ( zip[ 'model.obj' ] && zip[ 'materials.mtl' ] ) {
 
 			var { MTLLoader } = await import( '../../examples/jsm/loaders/MTLLoader.js' );
 			var { OBJLoader } = await import( '../../examples/jsm/loaders/OBJLoader.js' );
 
-			var materials = new MTLLoader().parse( zip.file( 'materials.mtl' ).asText() );
-			var object = new OBJLoader().setMaterials( materials ).parse( zip.file( 'model.obj' ).asText() );
+			var materials = new MTLLoader().parse( strFromU8( zip[ 'materials.mtl' ] ) );
+			var object = new OBJLoader().setMaterials( materials ).parse( strFromU8( zip[ 'model.obj' ] ) );
 			editor.execute( new AddObjectCommand( editor, object ) );
 
 		}
 
 		//
 
-		zip.filter( async function ( path, file ) {
```

Comment:
Was there a specific reason for using `filter()`? It seems it is just necessary to iterate through the object 🤔 .

Reply:
I don't remember 🤔